lib/FusionInventory/Agent/Task/Inventory/Win32/Firewall.pm

```diff
@@ -0,0 +1,201 @@
+
+use FusionInventory::Agent::Tools;
+use FusionInventory::Agent::Tools::Win32;
+use FusionInventory::Agent::Constants;
+
+use Storable 'dclone';
+
+my @mappingFirewallProfiles = qw/public standard domain/;
+
+sub isEnabled {
+    my (%params) = @_;
+    return 0 if $params{no_category}->{firewall};
+    return 1;
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my $profiles = _getFirewallProfiles();
+    my @profiles = _makeProfileAndConnectionsAssociation(firewallProfiles => $profiles, logger => $logger);
+    for my $profile (@profiles) {
+        $inventory->addEntry(
+            section => 'FIREWALL',
+            entry   => $profile
+        );
+    }
+
+}
+
+sub _getFirewallProfiles {
+
+    my $key = getRegistryKey( path =>
+        "HKEY_LOCAL_MACHINE/SYSTEM/CurrentControlSet/services/SharedAccess/Parameters/FirewallPolicy"
+    );
+    return unless $key;
+    my $subKeys = {
+        domain   => 'DomainProfile',
+        public   => 'PublicProfile',
+        standard => 'StandardProfile'
+    };
+    my $profiles = {};
+    for my $profile (keys %$subKeys) {
+        next unless $key->{$subKeys->{$profile} . '/'};
+        next unless defined $key->{$subKeys->{$profile} . '/'}->{'/EnableFirewall'};
+        my $enabled = hex2dec($key->{ $subKeys->{$profile} . '/'}->{'/EnableFirewall'});
+        $profiles->{$profile} = {
+            STATUS => $enabled
+                ? FIREWALL_STATUS_ON
+                : FIREWALL_STATUS_OFF,
+            PROFILE => $subKeys->{$profile}
+        };
+    }
+
+    return $profiles;
+}
+
+sub _makeProfileAndConnectionsAssociation {
+    my (%params) = @_;
+
+    return unless $params{firewallProfiles};
+
+    my ($profilesKey, $signaturesKey) = _retrieveProfilesAndSignaturesKey();
+    return unless $profilesKey && $signaturesKey;
+
+    foreach my $interface (getInterfaces(
+        additionalPropertiesNetWorkAdapterConfiguration => [qw/DNSDomain/],
+        additionalPropertiesNetWorkAdapter => [qw/GUID/]
+    )) {
+        next if ($interface->{STATUS} ne 'Up');
+
+        my $profile;
+
+        my $domainSettings = _getConnectionDomainSettings(
+            guid => $interface->{GUID}
+        );
+        # check if connection with domain
+        if ($domainSettings) {
+            $profile = _retrieveFirewallProfileWithdomain(
+                profileName => $domainSettings->{'/PrimaryDomainName'},
+                profilesKey => $profilesKey
+            )
+        } else {
+            $profile = _retrieveFirewallProfileWithoutDomain(
+                DNSDomain => $interface->{DNSDomain},
+                profilesKey => $profilesKey,
+                signaturesKey => $signaturesKey
+            );
+        }
+
+        next unless $profile;
+	
+        my $category = hex2dec($profile->{'/Category'});
+        if (not defined $params{firewallProfiles}->{$mappingFirewallProfiles[$category]}->{CONNECTIONS}) {
```

Comment:
You should prefer more perl-ish construct "unless (...)" in place of "if (not ...)"